packages/analytics-client-common/src/storage/cookie.ts

```diff
@@ -101,3 +101,21 @@ export class CookieStorage<T> implements Storage<T> {
     return;
   }
 }
+
+const decodeCookiesAsDefault = (value: string): string | undefined => {
```

Comment:
NIT: `decodeCookieWithoutURLEncoding` and `decodeCookieWithURLEncoding`?

default is a little too generic and the other one is not just for ruby but for any service that URL encodes cookies

Reply:
I'd like to keep `decodeCookiesAsDefault` because in this function value is still url decoded once. For `decodeCookiesForRuby`, how about `decodeCookiesByDoubleUrl`?